### ISSUE TITLE
`"@types/lucene": "^2"` becomes `"2.0.0"`, but that does not exist.

### ISSUE THREAD

**Issue by leeoniya** [NullVoxPopuli/defrag#8]
had to manually replace to `"@types/lucene": "2.1.7"`

**Comment by NullVoxPopuli** [comment]
thanks for reporting!

I have conflicting feelings about this one, because "the fix" would require confirring with the `npm` api for each package to see if the parsed/expanded version (from the `semver` package) actually exists. But that would _greatly_ increase the time that `defrag` takes to run. 

Maybe if the tool detects an interactive session, it can prompt for which version to use (as well as check npm) when the original version isn't fully specified (`^2`, `^2.x`, etc).

_hopefully_ this type of version specifying isn't all that common?

**Comment by leeoniya** [comment]
yeah some kind of iterative defrag -> install-> detect errors -> interactive selection of alternative versions -> defrag -> install...etc.

this way it's not necessary to manually edit versions in files and reference npm for whats available.

you can probably do this cycle auto to find latest ver that works.
